src/utils/__tests__/paging.test.ts

```diff
@@ -0,0 +1,94 @@
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import * as P from "../paging";
+import { NonNegativeInteger } from "@pagopa/ts-commons/lib/numbers";
+import {
+  flattenAsyncIterator} from "../async";
+
+const anIdBasedModel = {
+  id: "AAAAA" as NonEmptyString
+};
+
+const iteratorGenMock = async function*<T>(arr: T[]) {
+  for (let a of arr) yield a;
+};
+
+describe("Paging", () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it("should return a result page with the correct number of elements", async () => {
+    const pageSize = 3 as NonNegativeInteger;
+
+    const iteratorMock = iteratorGenMock([anIdBasedModel, anIdBasedModel, anIdBasedModel, {id: "BBBB" as NonEmptyString}]);
+    const results = await P.fillPage(iteratorMock as any, pageSize)
+    expect(results).toMatchObject(
+      {
+        hasMoreResults: true,
+        items: [anIdBasedModel, anIdBasedModel, anIdBasedModel],
+        next: anIdBasedModel.id,
+        prev: anIdBasedModel.id
+      }
+    );
+  });
+
+  it("should return an empty page if requested pageSize is 0", async () => {
+    const pageSize = 0 as NonNegativeInteger;
+
+    const iteratorMock = iteratorGenMock([anIdBasedModel, anIdBasedModel, anIdBasedModel, {id: "BBBB" as NonEmptyString}]);
```

Comment:
Why not having used my previous tests functions? There was a function that generated page of messages with a pending message every N. It was parameterized and allowed me to check also that fillPage correctly fills  a page with messages from different pages preserving the order.

Reply:
I'd rather to put these kind of tests in integration or even in the unit tests related to the client tag use this pagination utilities :)